### PR TITLE
API daemon 'getoffer' & 'gettrade' support for BSQ swaps #1

### PR DESCRIPTION
**Pull request by ghubstan**
This change is a partial adjustment in the API for BSQ swaps, mostly related to how it fetches and displays v1 and new swap offers and trades.

- The `currencyCode` param was removed from the `GetBsqSwapOffer` service request.

- `BsqSwapOfferInfo` fields were merged into `OfferInfo`, and `BsqSwapOfferInfo` was removed.

- New rpc service `GetOfferCategory` was added for later use, to determine what kind of  rpc request to send in CLI `takeoffer` calls (v1-offer or bsq-swap offer).
  
- Added new `coreApi.getRole(BsqSwapTrade)` method.

- Normalized (removed duplication) recently merged `BsqSwapTradeInfo` with the existing  `TradeInfo` gRPC proto defs and wrappers.  A `bsqSwapTradeInfo` field was appended to the old `TradeInfo` proto message.  The API's `gettrade`  and `gettrades` (todo) methods' reply messages can still return the same gRPC response message data type (`TradeInfo`, or `List<TradeInfo>` for `gettrades`), and may represent v1 trade and/or bisq-swap trade details.

- Adjusted the CLI console output to display v1 or bsq-swap trade details returned by the CLI's `gettrade` method.

BSQ swap offer create & take support in the CLI should come with the next 1-3 PRs.

Based on `master`.

### CODE REVIEW

**Comment by ripcurlx** [comment]
I tried `getoffer`, `getmyoffer`, `getoffers`, `getmyoffers`, `editoffer` and `gettrade`. `editoffer` fails as it is not implemented I guess, but `gettrade` also fails for bsq swap trades. Is this not to be expected to work in this PR as well?
```
./bisq-cli --password=mySecret gettrade --trade-id=411395-b5a74a47-ecfd-4111-9199-1c060261b281-175
ID      My Role              Price in BTC for 1 BSQ  Amount(BSQ)  Tx Fee (BTC)  Maker Fee(BSQ)  Deposit Published  Deposit Confirmed  Buyer Cost(BTC)  BSQ Sent  BSQ Received  Payout Published  Withdrawn  BSQ Buyer Address
411395  BSQ seller as maker              0.00004000       250.00    0.00002450            0.00  YES                NO                      0.01000000  NO        NO            NO                NO         Bbcrt1qu5tcy8kjmp09jdur78rq0djjar9hdla24rtlj5
./bisq-cli --password=mySecret gettrade --trade-id=9510828-952a1e3c-d658-4d9b-91ec-da75b830912e-175
Error: trade with id '9510828-952a1e3c-d658-4d9b-91ec-da75b830912e-175' not found
```

**Comment by ghubstan** [comment]
> I tried `getoffer`, `getmyoffer`, `getoffers`, `getmyoffers`, `editoffer` and `gettrade`. `editoffer` fails as it is not implemented I guess, but `gettrade` also fails for bsq swap trades. Is this not to be expected to work in this PR as well?

I have not addressed the CLI side of this, and do not expect it to work.  (The PR title should have specified these are incremental changes, mostly on the server side.)  This will require 2-3 more PRs.

I am a bit surprised that `gettrade` did not work, but again, I have not worked on the CLI side of `gettrade`.

Please un-comment the `@Disabled` annotation in `bisq.apitest.method.trade.BsqSwapTradeTest`, and run it with the `-DrunApiTests=true` jvm-opt?  That should work because of the changes to the related gRPC/protobuf defs and classes, and other changes on the 'server' side.  To see what `gettrade` _will_ show (in future PR), set `<root level="DEBUG">` in the apitest/test/resources/logback.xml.

File: core/src/main/java/bisq/core/offer/OfferUtil.java [hunk 1]
```diff
@@ -488,6 +488,14 @@ public OfferPayload getMergedOfferPayload(OpenOffer openOffer,
         }
     }
 
+    public static boolean isFiatOffer(Offer offer) {
+        return offer.getBaseCurrencyCode().equals("BTC") && !offer.isBsqSwapOffer();
+    }
+
+    public static boolean isAltcoinOffer(Offer offer) {
+        return offer.getCounterCurrencyCode().equals("BTC") && !offer.isBsqSwapOffer();
+    }
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/offer/OfferUtil.java:497]: Is swapOffer not considered an altcoin trade? I guess it makes some sense but also somewhat odd.

**Review by sqrrm — DISMISSED** [review]
utACK

@ghubstan I haven't followed other developments, can this be merged or does it need to wait?

**Pull request merged**